### DISCUSSION TITLE
Can history be loaded into the db?

### DISCUSSION THREAD

**Discussion by dkerr64** [jasonacox/Powerwall-Dashboard#212] [General]
I am just starting with Powerwall-Dashboard but have had Tesla solar and Powerwall for almost 2 years now.  I have been downloading historical data (each month, download CSV file with daily data from the iPhone app) and importing that into excel so I could then graph my generation, etc.

Is there any way to import historical data into Powerwall-Dashboard?  Or do I just need to be patient while is collects fresh data.

Thanks
David

**Comment by jasonacox** [comment]
Yes, and you won't need the data you manually collected. There is a great tool written by @mcbirse that imports all your historical data directly from Tesla. 

Here is the tool: https://github.com/jasonacox/Powerwall-Dashboard/tree/main/tools/tesla-history

**Comment by dkerr64** [comment]
This is awesome.  I managed to pull in all 2 years of history into Powerwall-dashboard.

Thank you
David

**Reply by mcbirse** [reply]
@dkerr64 - great to hear! Always nice to know when the script has been useful.
